### PR TITLE
Treatment for empty keys

### PR DESCRIPTION
**Pull request by alexandremsouza1**
Correcting the import problem with null value keys.

Added unit test that reproduces the problem.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,20 +57,31 @@ function importFromJsonString(idbDatabase, jsonString, cb) {
...
-      request.onsuccess = () => {
-        count++;
-        if (count === importObject[storeName].length) {
-          // added all objects for this store
-          delete importObject[storeName];
-          if (Object.keys(importObject).length === 0) {
-            // added all object stores
-            cb(null);
+    if (importObject[storeName] && Array.from(importObject[storeName]).length > 0) {
+      Array.from(importObject[storeName]).forEach((toAdd) => {
```
Comment (Polarisation) [inline index.js:61]: I think there should be a variable introduced to avoid repeating `Array.from(importObject[storeName])`

File: index.js [hunk 2]
```diff
@@ -16,7 +16,9 @@ function exportToJsonString(idbDatabase, cb) {
     );
     transaction.onerror = (event) => cb(event, null);
 
-    Array.from(idbDatabase.objectStoreNames).forEach((storeName) => {
+    const objectStoreNames = Array.from(new Set(idbDatabase.objectStoreNames));
```
Comment (Polarisation) [inline index.js:19]: Please could you explain how there can be duplicates?

File: test/test.js [hunk 3]
```diff
@@ -85,4 +87,37 @@ describe('IDBExportImport', function() {
       assert.ifError(e);
     });
   });
+  it('Should import and export the database with composite keys and empty keys', function(done) {
```
Comment (Polarisation) [inline test/test.js:90]: I don't understand what "composite keys" means here.

Does it refer to multiple stores? (colors, shapes, color_shape, empty)

Or does it refer to the data in the stores? Like `color_shape: '[shape+color]',`

File: index.js (hunk 1 above)
Comment (alexandremsouza1) [reply index.js]: Updated:  refactoring done

File: index.js (hunk 2 above)
Comment (alexandremsouza1) [reply index.js]: I think it's better to remove this part because  I couldn't reproduce the error

File: test/test.js (hunk 3 above)
Comment (alexandremsouza1) [reply test/test.js]: Improved the test description

File: index.js [hunk 4]
```diff
@@ -26,7 +28,7 @@ function exportToJsonString(idbDatabase, cb) {
         } else {
           exportObject[storeName] = allObjects;
           if (
-            idbDatabase.objectStoreNames.length ===
+            objectStoreNames.length ===
```
Comment (tiago-pullup) [inline index.js:31]: change this to objectStoreNames.size because a javascript Set don't have length in mobile devices

File: index.js [hunk 5]
```diff
@@ -16,7 +16,9 @@ function exportToJsonString(idbDatabase, cb) {
     );
     transaction.onerror = (event) => cb(event, null);
 
-    Array.from(idbDatabase.objectStoreNames).forEach((storeName) => {
+    const objectStoreNames = Array.from(idbDatabase.objectStoreNames);
```
Comment (tiago-pullup) [inline index.js:19]: change Array.from(idbDatabase.objectStoreNames) to Set(idbDatabase.objectStoreNames) avoid duplications store names

File: index.js (hunk 4 above)
Comment (tiago-pullup) [inline index.js:31]: change this to objectStoreNames.size because a javascript Set don't have length in mobile devices

File: index.js (hunk 5 above)
Comment (alexandremsouza1) [reply index.js]: Done

File: index.js (hunk 4 above)
Comment (alexandremsouza1) [reply index.js:31]: Done

Comment (alexandremsouza1) [reply index.js:31]: the changes were made because using the **Apache Cordova** for mobile devices happens these problems.

**Pull request merged**